### DISCUSSION TITLE
Ability to execute composer update and install via a UI

### DISCUSSION THREAD

**Discussion by Wulfheart** [leokhoa/laragon#586] [Ideas]
Hey,

I introduced Laragon to three Newbies yesterday as a simple way to get started with PHP development on Windows. I think it would be great if there was a possibility to allow them to execute composer/console commands via a graphical UI.

**Comment by RobiNN1** [comment]
In development, a lot of things are done via the cli ;). IMHO sometimes the cli is much faster than finding the same thing in UI.

But there are tools like [Composercat](https://getcomposercat.com/) that provide GUI for Composer. I've tried it but again cli is better and faster.

**Comment by Pen-y-Fan** [comment]
PHPStorm has links to **install** or **update** when **composer.json** is opened.

<img width="889" alt="image" src="https://github.com/leokhoa/laragon/assets/40126936/bb92c388-6b62-4356-9ed5-9414c2c37fda">

They run `composer install` or `composer update` from the command line for you.

You can also press CTRL twice to bring up **Run anything** and type or paste in a composer command, this will run the command from the command line for you. This is sometimes useful with artisan commands, e.g. if I wanted to create an OrderResource and a UserResource, it's easy to type it once and run it.  Then bring up the Run anything again, down arrow to select the previous command, then edit the command, CTRL left to move the carat to the previous word. Change Order to User. Terminals can be hit and miss when trying this.

Most of the time I use CTRL TAB T to bring up the terminal and type the command there, as any interactive commands will need to be run from the command line anyway.

**Reply by RobiNN1** [reply]
I don't think newbie have PHPStorm. He probably uses that slow crap called VSCode, maybe there is a plugin for that.

But yes, PHPStorm has many useful features like this.
